music21/tinyNotation.py

```diff
@@ -1256,6 +1375,56 @@ def testOne(self):
         self.assertEqual(sfn[12].duration.quarterLength, 1.0)
         self.assertEqual(sfn[12].expressions[0].classes, expressions.Fermata().classes)
 
+    def testRaiseExceptions(self):
+        error_states = [
+            {
+                "string": "h",
+                "reason": "h is not a valid note",
+            },
+            {
+                "string": "a;",
+                "reason": "a semicolon is not a valid character or modifier",
+            },
+            {
+                "string": "r;",
+                "reason": "a semicolon is not a valid character or modifier",
+            },
+            {
+                "string": "4/4;",
+                "reason": "a semicolon is not a valid character or modifier",
+            },
+            {
+                "string": "ABC",
+                "reason": (
+                    "only the same upper-cased letter may be repeated to "
+                    "indicate lower octaves"
+                ),
+            },
+        ]
+
+        for error_state in error_states:
+            try:
+                converter = Converter(error_state["string"], raiseExceptions=True)
+                result = converter.parse()
+                result.stream.show('text')
+                self.fail(
+                    "Should have raised a TinyNotationException for input "
+                    f'"{error_state["string"]}" because {error_state["reason"]}.'
+                )
+            except TinyNotationException as e:
+                # We expected to raise this exception
+                pass
+            except AssertionError:
+                # The test failed. Reraise the original test failure
+                raise
+            except Exception as e:
+                raise AssertionError(
+                    "Should have raised a TinyNotationException for input "
+                    f'"{error_state["string"]}" because {error_state["reason"]} '
+                    f"but actually raised {e.__class__.__name__}."
+                ) from e
```

Comment:
Can we just use `with self.assertRaises(TinyNotationException):` here? Shorter, consistent with the rest of the suite, and avoids missed lines in the Coveralls report.

Reply:
Yeah, we can do that. The only reason I went with this approach is that I wanted to leave a more descriptive error message if the test fails.

Reply:
I'll just reraise with additional information. That should be simpler

Reply:
Oh wait! Today I learned that they've updated `assertRaises` to accept a `msg` parameter =D
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises